test/plugin/test_out_forward.rb

```diff
@@ -621,6 +624,105 @@ def try_write(chunk)
+    root_ca = root[0]
+
+    root_ca_file = Tempfile.new("root-cert")
+    root_ca_file.write(root_ca.to_pem())
+    root_ca_file.close
+    # No need to write root key to file
+
+    intermediate = create_certificate("intermediate-ca", 2, root)
+    # The intermediate_ca is not written to a file - it will be in the
+    # client and service certificates' chain.
+
+    # Test both (root->cert) and (root->intermediate->cert)
+    parents = [root, intermediate]
+    parents.each do |parent|
+      client = create_certificate("client-cert", 3, parent, false)
+      client_cert = client[0]
+      client_key = client[1]
+      client_cert_file = Tempfile.new("client-cert")
+      client_cert_file.write(client_cert.to_pem())
+      if parent != root
+        client_cert_file.write(parent[0].to_pem())
+      end
+      client_cert_file.close
+
+      client_key_file = Tempfile.new("client-key")
+      client_key_file.write(client_key.to_pem())
+      client_key_file.close
+
+      server = create_certificate("server-cert", 4, parent, false)
+      server_cert = server[0]
+      server_key = server[1]
+
+      server_cert_file = Tempfile.new("server-cert")
+      server_cert_file.write(server_cert.to_pem())
+      if parent != root
+        server_cert_file.write(parent[0].to_pem())
+      end
+      server_cert_file.close
+
+      server_key_file = Tempfile.new("server-key")
+      server_key_file.write(server_key.to_pem())
+      server_key_file.close
+
+      input_conf = TARGET_CONFIG + %[
+                     <transport tls>
+                       cert_path #{server_cert_file.path}
+                       private_key_path #{server_key_file.path}
+                       client_cert_auth true
+                       ca_path #{root_ca_file.path}
+                     </transport>
+                   ]
+      target_input_driver = create_target_input_driver(conf: input_conf)
+
+      output_conf = %[
+        send_timeout 5
+        require_ack_response #{ack}
+        transport tls
+        tls_verify_hostname false
+        tls_cert_path #{root_ca_file.path}
+        tls_client_cert_path #{client_cert_file.path}
+        tls_client_private_key_path #{client_key_file.path}
+        <server>
+          host #{TARGET_HOST}
+          port #{TARGET_PORT}
+        </server>
+        <buffer>
+          #flush_mode immediate
+          flush_interval 0s
+          flush_at_shutdown false # suppress errors in d.instance_shutdown
+        </buffer>
+      ]
+      @d = d = create_driver(output_conf)
+
+      time = event_time("2011-01-02 13:14:15 UTC")
+      records = [{"a" => 1}, {"a" => 2}]
+      target_input_driver.run(expect_records: 2, timeout: 3) do
+        d.run(default_tag: 'test', wait_flush_completion: false, shutdown: false) do
+          records.each do |record|
+            d.feed(time, record)
+          end
+        end
+      end
+
+      events = target_input_driver.events
+      assert{ events != [] }
```

Comment:
`assert` has little information when this case failed. so how about this?

```suggestion
      assert_not_equal(event, [])
```

Reply:
will do - note that this is copied from the test above though